### PR TITLE
[1.12]added CriticalHitEvent, closes #2871

### LINKED ISSUE #2871: [Suggestion] Event to handle critical hits
In 1.8.9 it is not possible to determin if a player make a critical hit nor prevent him form doing one. An external event for handling it whould be very nice. We could also add an event for handling critical enchantment hits.

If I am allowed, then I can make a pull request. #

### PR DESCRIPTION
**Pull request by mcenderdragon**
Retarget of #3190, and fixes #2871

### CODE REVIEW

File: src/main/java/net/minecraftforge/common/ForgeHooks.java [hunk 1]
```diff
@@ -1259,6 +1260,18 @@ public static boolean loadAdvancements(Map<ResourceLocation, Advancement.Builder
         return errored;
     }
 
+    @Nullable
+    public static CriticalHitEvent getCriticalHit(EntityPlayer player, Entity target, boolean vanillaCritical, float damageModifier)
+    {
+        CriticalHitEvent hitResult = new CriticalHitEvent(player, target, damageModifier);
```
Comment (liach) [inline src/main/java/net/minecraftforge/common/ForgeHooks.java]: I'd prefer that mods can access vanilla results.

Comment (mcenderdragon) [reply src/main/java/net/minecraftforge/common/ForgeHooks.java]: They can.

Comment (mezz) [reply src/main/java/net/minecraftforge/common/ForgeHooks.java]: I think he is saying `CriticalHitEvent` should take in `vanillaCritical` as well.

**Comment by mezz** [comment]
Looks good, please resolve conflicts and liach's request.

File: src/main/java/net/minecraftforge/common/ForgeHooks.java (hunk 1 above)
Comment (mcenderdragon) [reply src/main/java/net/minecraftforge/common/ForgeHooks.java]: Ohh, my bad. Yeah I'll fix it.

**Comment by mcenderdragon** [comment]
Ok it should be fixed now.

**Review by liach — APPROVED** [review]

**Comment by mezz** [comment]
Tested and looks good to me.

**Pull request merged**